src/scripts/main.js

```diff
@@ -1,3 +1,57 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tbody = table.querySelector('tbody');
+const keys = ['name', 'position', 'age', 'salary'];
+
+const data = Array.from(tbody.rows).map((row) => {
+  const cells = Array.from(row.cells);
+  const object = {};
+
+  cells.forEach((cell, index) => {
+    object[keys[index]] = cell.textContent;
+  });
+
+  return object;
+});
+
+function getSorted() {
+  tbody.innerHTML = '';
+
+  data.forEach((row) => {
+    const tr = document.createElement('tr');
+
+    Object.values(row).forEach((cellContent) => {
+      const td = document.createElement('td');
+
+      td.textContent = cellContent;
+      tr.appendChild(td);
+    });
+
+    tbody.appendChild(tr);
+  });
+}
+
+table.addEventListener('click', (e) => {
+  if (e.target === table.rows[0].cells[0]) {
```

Comment:
Consider checking if the clicked element is indeed a header cell. Currently, it assumes the first row contains headers, which might not always be the case.